### PR TITLE
yices: new port in math

### PR DESCRIPTION
**Pull request by barracuda156**
#### Description

New port in math: https://yices.csl.sri.com

_PPC support is experimental. At the moment larger part of tests fail on ppc32._

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.6.8 Server
Xcode 3.2.6

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by barracuda156** [comment]
Hmm, another relocation error even on x86_64:
```
ld: warning: disabling chained fixups because of unaligned pointers
[2349](https://github.com/macports/macports-ports/actions/runs/3789041526/jobs/6442415011#step:8:2352)
  ld: illegal text-relocation to '___gmp_binvert_limb_table' in /opt/local/lib/libgmp.a(mp_minv_tab.o) from '___gmpn_divexact_1' in /opt/local/lib/libgmp.a(dive_1.o) for architecture x86_64
[2350](https://github.com/macports/macports-ports/actions/runs/3789041526/jobs/6442415011#step:8:2353)
  clang: error: linker command failed with exit code 1 (use -v to see invocation)
[2351](https://github.com/macports/macports-ports/actions/runs/3789041526/jobs/6442415011#step:8:2354)
  make[2]: *** [../build/x86_64-apple-darwin21.6.0-release/static_lib/libyices.2.dylib] Error 1
[2352](https://github.com/macports/macports-ports/actions/runs/3789041526/jobs/6442415011#step:8:2355)
  make[2]: *** Waiting for unfinished jobs....
[2353](https://github.com/macports/macports-ports/actions/runs/3789041526/jobs/6442415011#step:8:2356)
  ranlib ../build/x86_64-apple-darwin21.6.0-release/static_lib/libyices.a
[2354](https://github.com/macports/macports-ports/actions/runs/3789041526/jobs/6442415011#step:8:2357)
  /Library/Developer/CommandLineTools/usr/bin/ranlib: file: ../build/x86_64-apple-darwin21.6.0-release/static_lib/libyices.a(yices_locks.o) has no symbols
[2355](https://github.com/macports/macports-ports/actions/runs/3789041526/jobs/6442415011#step:8:2358)
  make[2]: Leaving directory `/opt/local/var/macports/build/_Users_runner_work_macports-ports_macports-ports_ports_math_yices/yices/work/SRI-CSL-yices2-f705557/src'
[2356](https://github.com/macports/macports-ports/actions/runs/3789041526/jobs/6442415011#step:8:2359)
  make[1]: *** [static-lib] Error 2
```
Amazingly written code LOL.
Let me fix that.

**Comment by barracuda156** [comment]
A quick search showed that this error results from a bug in `gmp`, and we had a ticket for that, but without a fix: https://trac.macports.org/ticket/53361

Related reports:
https://gmplib.org/list-archives/gmp-discuss/2017-September/006145.html
https://mail-index.netbsd.org/pkgsrc-users/2015/07/10/msg021819.html
https://github.com/r-hub/homebrew-cran/issues/3

**Comment by barracuda156** [comment]
I removed static targets from all, now the port builds fine. Locally ppc32 build also worked.

File: math/yices/Portfile [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+PortGroup           github 1.0
+
+github.setup        SRI-CSL yices2 2.6.4 Yices-
+name                yices
+revision            0
+categories          math
+license             GPL-3
+maintainers         {@barracuda156 gmail.com:vital.had} openmaintainer
+description         The Yices SMT Solver
+long_description    ${description}
```
Comment (herbygillot) [inline math/yices/Portfile:14]: Same issue with `${description}` as in the `libfixmath` PR: reusing `${description}` requires the Tcl expansion operator (`{*}`).

```suggestion
long_description    {*}${description}
```

Comment (barracuda156) [reply math/yices/Portfile:14]: Sorry, forgot to update this. Will fix it now.

**Pull request merged**